lib/ansible/modules/cloud/amazon/dynamodb.py

```diff
@@ -0,0 +1,359 @@
+        table=dict(required=True),
+        action=dict(required=True, choices=["get", "put", "update", "delete"]),
+        primary_key=dict(required=False, type='dict'),
+        item=dict(required=False, type='dict'),
+        condition_expression=dict(required=False, type='str'),
+        expression_attribute_names=dict(required=False, type='dict'),
+        expression_attribute_values=dict(required=False, type='dict'),
+        update_expression=dict(required=False, type='str'),
+        projection_expression=dict(required=False, type='str')
+    )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='pip modules boto3 and botocore are required')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = ansible.module_utils.ec2.get_aws_connection_info(
+            module, boto3=True)
+        connection = ansible.module_utils.ec2.boto3_conn(module, conn_type='client',
+                                                         resource='dynamodb',
+                                                         region=region, endpoint=ec2_url,
+                                                         **aws_connect_kwargs)
+    except (botocore.exceptions.NoRegionError,
+            botocore.exceptions.ProfileNotFound,
+            botocore.exceptions.NoCredentialsError) as error:
+
+        error_msg = boto_exception(error)
+        module.fail_json(msg=error_msg)
+
+    try:
+        table = module.params.get('table')
+        action = module.params.get('action')
+        primary_key = module.params.get('primary_key')
+        condition_expression = module.params.get('condition_expression')
+        item = module.params.get('item')
+        update_expression = module.params.get('update_expression')
+        expression_attribute_names = module.params.get(
+            'expression_attribute_names')
+        expression_attribute_values = module.params.get(
+            'expression_attribute_values')
+        projection_expression = module.params.get('projection_expression')
+
+        result = dict(
+            changed=False,
+            item=''
+        )
+
+        if action == 'get':
+            result = get(connection, table, primary_key, projection_expression,
+                         result)
+
+        elif action == 'put':
+            result = put(connection, table, item, result)
+
+        elif action == 'update':
+            result = update(connection, table, primary_key, update_expression,
+                            condition_expression, expression_attribute_names,
+                            expression_attribute_values, result)
+
+        elif action == 'delete':
+            result = delete(connection, table, primary_key,
+                            condition_expression, expression_attribute_values,
+                            result)
+
+    except connection.exceptions.ResourceNotFoundException as error:
```

Comment:
Exception handling should probably be done closer to the actual boto calls (e.g. in `get()`), and it should definitely use `fail_json_aws()` and catch `BotoCoreError` and `ClientError` instead of `Exception`.
```
except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
    module.fail_json_aws(e, msg='Failed to get item')
```